### PR TITLE
[ODS-5459] Create AuthorizationContextDataProvider for role-named ResponsibilitySchoolId in Discipline Action

### PR DESCRIPTION
**Pull request by ericstockdale**

### CODE REVIEW

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationContextData.cs [hunk 1]
```diff
@@ -28,6 +28,7 @@ private static class PropertyNames
             public const string StaffUSI = "StaffUSI";
             public const string StudentUSI = "StudentUSI";
             public const string ParentUSI = "ParentUSI";
+            public const string ResponsibilitySchoolId = "ResponsibilitySchoolId";
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationContextData.cs:31]: This class only contains entries for the baseline EdOrg-derived properties. Role named properties are not "new" properties, they are contextual variations of the main ones. There should be no changes to this class.

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationContextData.cs [hunk 2]
```diff
@@ -41,6 +42,8 @@ private static class PropertyNames
 
         public int? SchoolId { get; set; }
 
+        public int? ResponsibilitySchoolId { get; set; }
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationContextData.cs:45]: Remove this.

File: Application/EdFi.Ods.Standard/Security/Authorization/Overrides/DisciplineActionRelationshipsAuthorizationContextDataProvider.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        public TContextData GetContextData(IDisciplineAction resource)
+        {
+            if (resource == null)
+                throw new ArgumentNullException("resource", "The 'disciplineAction' resource for obtaining authorization context data cannot be null.");
+
+            var entity = resource as DisciplineAction;
+
+            var contextData = new TContextData();
+            // AssignmentSchoolId = entity.AssignmentSchoolId, // Role name applied and not part of primary key
+            contextData.ResponsibilitySchoolId = entity.ResponsibilitySchoolId; // Role name applied and not part of primary key
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Standard/Security/Authorization/Overrides/DisciplineActionRelationshipsAuthorizationContextDataProvider.cs:33]: This should be:
```csharp
contextData.SchoolId = entity.ResponsibilitySchoolId; // Role name applied and not part of primary key
```

**Review by gmcelhanon — CHANGES_REQUESTED** [review]
Make the changes listed and let's see where we are.

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationContextData.cs (hunk 2 above)
Comment (semalaiappan) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationContextData.cs]: OK Thanks  .  I removed it .

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationContextData.cs [hunk 4]
```diff
@@ -101,7 +104,10 @@ private static class PropertyNames
                     yield return (authorizationContextPropertyName, SchoolId);
 
                     break;
+                case PropertyNames.ResponsibilitySchoolId:
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationContextData.cs]: I removed it .

**Pull request closed without merging**